spanner/api/Spanner.Samples.Tests/UpdateDatabaseWithDefaultLeaderAsyncTest.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Spanner.Admin.Database.V1;
+using Google.Cloud.Spanner.Common.V1;
+using System.Threading.Tasks;
+using Xunit;
+
+[Collection(nameof(SpannerFixture))]
+public class UpdateDatabaseWithDefaultLeaderAsyncTest
+{
+    private readonly SpannerFixture _spannerFixture;
+
+    public UpdateDatabaseWithDefaultLeaderAsyncTest(SpannerFixture spannerFixture)
+    {
+        _spannerFixture = spannerFixture;
+    }
+
+    [Fact]
+    public async Task TestUpdateDatabaseWithDefaultLeaderAsync()
+    {
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(_spannerFixture.InstanceIdWithMultiRegion, async databaseId =>
```

Comment:
(See also my comments above on the `SpannerFixture`)

The tradeoff here is that:
1. Having a fixed database on the multi-region instance would mean that for this test to actually do something, we would need to first set the default leader to X, and then let the sample change it to Y. Executing two separate `ALTER DATABASE` statements on an existing database actually takes longer (49.1 sec) than executing one `CreateDatabase` RPC with an `ALTER DATABASE` statement in the `ExtraStatements` list and then one `ALTER DATABASE` statement (30.6 sec).
2. By not using a fixed database that is tied to `SpannerFixture`, it would be easier to make this test independent of `SpannerFixture`, which again would make it quicker to run as a standalone test.